### PR TITLE
Add ability to use full AuthInfo in CustomAuthMethodInfo. 

### PR DESCRIPTION
**Pull request by hoerup**
Copy of https://github.com/rajanadar/VaultSharp/pull/81/

This enables to renew token based on lease duration and other information that is returned in scope of token issue request.

The use-case is next. We have CustomAuthMethodInfo and want to renew token, but renew is done in time authInfo.LeaseDurationInSeconds / 2 and using renew-self method.
In case if just token implementation is used there is no ReturnedLoginAuthInfo which can be used to fetch information about token and be able to effectively renew token

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+version: '1.0.{build}'
```
Comment (Marusyk) [inline appveyor.yml]: Do we need this file?

File: src/VaultSharp/AssemblyConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+using System.Runtime.CompilerServices;
```
Comment (Marusyk) [inline src/VaultSharp/AssemblyConfiguration.cs]: Also not sure about this. Can we use simple
`[assembly: InternalsVisibleTo("VaultSharp.UnitTest")]` in any of other file instead adding AssemblyConfiguration?

File: appveyor.yml (hunk 1 above)
Comment (hoerup) [reply appveyor.yml]: I actually prefer having CI config as file/code so it is under version control as well. Just never got round to set it up

File: src/VaultSharp/AssemblyConfiguration.cs (hunk 2 above)
Comment (hoerup) [reply src/VaultSharp/AssemblyConfiguration.cs]: It *could* go into VaultClient.cs but I like the idea of having it in a seperate file

**Pull request merged**